alexaCapabilities.js

```diff
@@ -95,29 +95,33 @@ AlexaCapabilities.prototype.percentageController = function () {
   };
 };
 
-AlexaCapabilities.prototype.thermostatController = function (targetSetpoint, upperSetpoint, lowerSetpoint, thermostatMode, supportedModes) {
+AlexaCapabilities.prototype.thermostatController = function (targetSetpoint, upperSetpoint, lowerSetpoint, thermostatMode) {
   var supported = [];
   if (targetSetpoint) {
     supported.push({
       "name": "targetSetpoint"
-    })
+    });
   }
   if (upperSetpoint) {
     supported.push({
       "name": "upperSetpoint"
-    })
+    });
   }
   if (lowerSetpoint) {
     supported.push({
       "name": "lowerSetpoint"
-    })
+    });
   }
   if (thermostatMode) {
     supported.push({
       "name": "thermostatMode"
-    })
+    });
+  }
+  var configuration = {};
+  if (typeof thermostatMode.parameters.supportedModes === 'string') {
```

Comment:
this is throwing an error if variables are not defined, something like this to fix:
```javascript
if (thermostatMode && thermostatMode.parameters && typeof thermostatMode.parameters.supportedModes === 'string') {
    configuration.supportedModes = thermostatMode.parameters.supportedModes.split(',').map(mode => mode.trim());
  }
```

Reply:
Thanks. I forgot the fact that a thermostat mode property may not always be defined however no need to check for parameters. If the property exists than it will have a parameter object defined.

https://github.com/jsetton/openhab-alexa/blob/v3-changes/alexaPropertyMap.js#L126

I will make the change in my local repository and wait for other changes that may found.

```js
if (thermostatMode && typeof thermostatMode.parameters.supportedModes === 'string') {
    configuration.supportedModes = thermostatMode.parameters.supportedModes.split(',').map(mode => mode.trim());
}
```